feat: setup /learning-marathon/[id] page

ruby10127130 opened the pull request:
新增
- 學習馬拉松： /learning-marathon/[id] 頁面，後續會作為獨立編輯頁（報名用戶回填 / 展示頁）

ruby10127130 committed:
feat: setup /learning-marathon/[id] page

JohnsonMao commented on pages/learning-marathon/[id]/index.jsx:
```diff
@@ -0,0 +1,521 @@
+const Panel = ({ children, className = "" }) => {
+  return (
+    <div className={
+      classNames('w-full sm:w-[750px] mx-auto rounded-2xl p-10', className)
```
這邊請使用
```
import { cn } from '@/utils/cn';

cn('xxx', className);
```
來合併 tailwindCSS 的 className

classnames 並沒有提供去掉同類型 tailwindCSS 樣式的功能

我們專案中也沒有直接安裝 classnames 的套件ＸＤ

JohnsonMao commented on pages/learning-marathon/[id]/index.jsx:
```diff
@@ -0,0 +1,521 @@
+          </h3>
+        )}
+    </>
+  );
```
這邊程式碼能幫我 format 成
```ts
return isLoading ? (
    <Skeleton animation="wave" width="60%" height="28px" />
  ) : (
    <h3 className="text-basic-500 body-md font-medium mb-2 font-sans">
      {title}
    </h3>
  );
```

另外 loading 能幫我命名成 isLoading，我個人比較習慣用 is 開頭來命名布林值的變數

JohnsonMao commented on pages/learning-marathon/[id]/index.jsx:
```diff
@@ -0,0 +1,521 @@
+  return (
+    <>
+      <hr className="my-5 border-basic-100" />
+    </>
```
這裡 <></> 似乎多此一舉～

JohnsonMao commented on pages/learning-marathon/[id]/index.jsx:
```diff
@@ -0,0 +1,521 @@
+        <p className="body-md font-sans">{description}</p>
+      )}
+    </>
+  );
```
同 Title 的 coding style

JohnsonMao commented on pages/learning-marathon/[id]/index.jsx:
```diff
@@ -0,0 +1,521 @@
+        </div>
+      )}
+    </>
+  );
```
同 Title 的 coding style

JohnsonMao commented on pages/learning-marathon/[id]/index.jsx:
```diff
@@ -0,0 +1,521 @@
+        </div>
+      )}
+    </>
+  );
```
同 Title 的 coding style

JohnsonMao commented on pages/learning-marathon/[id]/index.jsx:
```diff
@@ -0,0 +1,521 @@
+            }
+          </div>
+        )
+      }
```
這裡有嵌套的三元運算子，有可讀性更好的撰寫方式嗎～？
另外一樣 isLoading 的命名方式幫我調整～

JohnsonMao commented on pages/learning-marathon/[id]/index.jsx:
```diff
@@ -0,0 +1,521 @@
+  const weekDay = subMilestone.dates.map((isoDate) => {
+    return ISOToWeekday(isoDate);
+  });
+  return (
+    <>
+      <div className="w-full bg-white rounded-lg py-3 px-4">
+        <p className="mb-2">
+          {`${index + 1}. `}{subMilestone.name}
+        </p>
+        <div className="flex flex-row items-center justify-start">
+          <CalendarTodayIcon
+            className="text-basic-300 mr-[5px]"
+            sx={{
+              width: '16px',
+              height: '16px'
+            }}
+          />
+          <p className="font-sans text-basic-300 text-xs font-light">
+            {weekDay.join(', ')}
+          </p>
+        </div>
+      </div>
+    </>
+  );
```
這裡的 <></> 似乎多此一舉

JohnsonMao reviewed (changes requested):
在麻煩調整一下～
另外不確定現在的架構有沒有支援 SSR
因為我們 redux 相依 client 端的套件還沒有調整ＸＤ

JohnsonMao commented on pages/learning-marathon/[id]/index.jsx:
```diff
@@ -0,0 +1,521 @@
+  const [loadingUser, setLoadingUser] = useState(false);
+
+  // set url for social share
+  useEffect(() => {
+    if ((typeof window !== 'undefined') && !loadedUrl) {
+      const currentUrl = window.location.origin + router.asPath;
+      setfullUrl(currentUrl);
+      setLoadedUrl(true);
+    }
+  }, [router]);
```
這邊應該只要撰寫
```ts
  useEffect(() => {
    setFullUrl(window.location.href);
  }, []);
  ```
  
  另外 setfullUrl 應該要改成 setFullUrl，小駝峰命名規則

ruby10127130 committed:
refactor: replace classnames with /utils/cn

ruby10127130 committed:
chore: remove un used empty fragment and variables

ruby10127130 committed:
chore: rename variable and function for consistency
- rename 'loading' to 'isLoading'
- rename setfullUrl to setFullUrl for camelCase consistency

ruby10127130 committed:
refactor: simplify conditional rendering

ruby10127130 committed:
refactor: simplify setfullUrl in useEffect

ruby10127130 commented:
> 不確定現在的架構有沒有支援 SSR
> 因為我們 redux 相依 client 端的套件還沒有調整ＸＤ

@JohnsonMao 目前 local 開得起來，這個頁面中也沒有使用 `useSWR` 或是 `getServerSideProps`，我想目前架構未轉換 SSR 應該不影響⋯⋯？

JohnsonMao commented:
因為路徑命名使用 動態路徑，所以這頁已經轉成 SSR
同時又有使用 redux 所以結果可能會跟開發環境不一致，我們來試試看XD
